core/runtime/src/main/java/io/quarkiverse/langchain4j/ChatMemoryRemover.java

```diff
@@ -0,0 +1,39 @@
+package io.quarkiverse.langchain4j;
+
+import java.util.List;
+
+import dev.langchain4j.memory.ChatMemory;
+import dev.langchain4j.store.memory.chat.ChatMemoryStore;
+import io.quarkiverse.langchain4j.runtime.aiservice.ChatMemoryRemovable;
+
+/**
+ * Allows the application to manually control when a {@link ChatMemory} should be removed from the underlying
+ * {@link ChatMemoryStore}.
+ */
+public final class ChatMemoryRemover {
+
+    private static final Object[] EMPTY_OBJECT_ARRAY = new Object[0];
+
+    private ChatMemoryRemover() {
+    }
+
+    /**
+     * @param aiService The bean that implements the AI Service annotated with {@link RegisterAiService}
+     * @param memoryId The object used as memory IDs for which the corresponding {@link ChatMemory} should be removed
+     */
+    public static void remove(Object aiService, Object memoryId) {
```

Comment:
I don't like using two objects here as users could easily swap them, but I don't have a better way of handling it...